eng/pipelines/templates/build-extension.yml

```diff
@@ -0,0 +1,149 @@
+parameters:
+  - name: buildConfig
+    type: string
+  - name: repoArtifactsPath
+    type: string
+  - name: repoLogPath
+    type: string
+  - name: buildArgs
+    type: string
+    default: ''
+
+steps:
+  - task: NodeTool@0
+    displayName: 🟣Install node.js
+    inputs:
+      versionSpec: '20.x'
+
+  - task: PowerShell@2
+    displayName: 🟣Install yarn
+    inputs:
+      targetType: 'inline'
+      script: |
+        npm install -g yarn
+        yarn --version
+      workingDirectory: '$(Build.SourcesDirectory)'
+
+  - task: PowerShell@2
+    displayName: 🟣Install vsce
+    inputs:
+      targetType: 'inline'
+      script: |
+        npm install -g @vscode/vsce
+        vsce --version
+      workingDirectory: '$(Build.SourcesDirectory)'
+
+  - task: PowerShell@2
+    displayName: 🟣Install dependencies and compile
+    inputs:
+      targetType: 'inline'
+      script: |
+        # Install dependencies
+        yarn install
+
+        # Compile
+        yarn compile
+      workingDirectory: '$(Build.SourcesDirectory)/extension'
+      errorActionPreference: 'stop'
+    env:
+      YARN_CACHE_FOLDER: $(Pipeline.Workspace)/.yarn
+
+  - task: PowerShell@2
+    displayName: 🟣Package extension as VSIX
```

Comment:
Can we move this to a msbuild target, so it isn't specific to this azdo pipeline? Maybe we can add a new parameter like `-build-extension` to generate the vsix. `-build-installers`?

- Once it is a msbuild target then we can build it from the main windows job itself.
- install the node dependencies on that job.
- And use `-build-extension -build-installers` on the build command
- And the signing would also automatically work
- and we can drop this file

Please correct me if I'm missing something.